### PR TITLE
Add splunk logging

### PR DESCRIPTION
**Pull request by atoulme**
This adds splunk logging to Besu.

Splunk logging is added as a runtime jar that is pulled from the Spring releases Maven repository.
This PR also removes the Maven repository for the ASF snapshots repository, as no dependencies use this repository at this time.

This change allows users to define Splunk as logger using system properties or environment variables.
This also allows users to change the log level using environment variables if they so choose.

To configure Splunk, you may set the following environment variables:

| Name             | Description    | Required       |
| ------------- | ------------- |------------- |
| LOGGER | Must be set to `Splunk` to activate Splunk logging | Yes |
| HOST | Current host. If in a Docker environment, the default value is the docker container ID. Otherwise the default value is `localhost`. | No |
| SPLUNK_URL | URL to the Splunk instance, for example https://localhost:8088 | Yes |
| SPLUNK_TOKEN | Authentication token, usually of the form 11111111-1111-1111-1111-111111111111 | Yes |
| SPLUNK_INDEX | [Index](https://docs.splunk.com/Splexicon:Index) in which logs will be stored. Defaults to `besu` | Yes |
| SPLUNK_SOURCE | [Source of the logs](https://docs.splunk.com/Splexicon:Source). This may be overridden by the Splunk HTTP Event Collector. Defaults to `besu` | No |
| SPLUNK_SOURCETYPE | [Sourcetype of the logs](https://docs.splunk.com/Splexicon:Sourcetype). This may be overridden by the Splunk HTTP Event Collector. Defaults to `besu` | No |
| SPLUNK_BATCH_SIZE_BYTES | Size of a batch in bytes. Defaults to `65536` | No |
| SPLUNK_BATCH_SIZE_COUNT | Size of a batch in number of events. Defaults to `1000` | No |
| SPLUNK_BATCH_INTERVAL | Interval at which to send batches. Defaults to `500` | No |
| SPLUNK_SKIPTLSVERIFY | Whether to check the Splunk instance certificate when sending data. Defaults to `true` | No |

Screen Shot 2020-04-14 at 3 45 45 PM

### CODE REVIEW

File: besu/src/main/resources/log4j2.xml [hunk 1]
```diff
@@ -1,16 +1,50 @@
...
+    <Property name="root.log.level">${env:LOG_LEVEL:-INFO}</Property>
+    <Property name="root.log.logger">${env:LOGGER:-Console}</Property>
+    <Property name="host">${env:HOST:-${docker:containerId:-localhost}}</Property>
+    <Property name="splunk.url">${env:SPLUNK_URL}</Property>
+    <Property name="splunk.token">${env:SPLUNK_TOKEN}</Property>
+    <Property name="splunk.index">${env:SPLUNK_INDEX}</Property>
+    <Property name="splunk.source">${env:SPLUNK_SOURCE:-besu}</Property>
+    <Property name="splunk.sourcetype">${env:SPLUNK_SOURCETYPE:-besu}</Property>
+    <Property name="splunk.batch_size_bytes">${env:SPLUNK_BATCH_SIZE_BYTES:-10000}</Property>
+    <Property name="splunk.batch_size_count">${env:SPLUNK_BATCH_SIZE_COUNT:-10}</Property>
```
Comment (ziegfried) [inline besu/src/main/resources/log4j2.xml]: 10 seem quite low for a default batch size.

Comment (atoulme) [reply besu/src/main/resources/log4j2.xml]: I made up those numbers. Does 100 sound good?

Comment (atoulme) [reply besu/src/main/resources/log4j2.xml]: I pushed to 100 for now.

File: besu/src/main/resources/log4j2.xml [hunk 2]
```diff
@@ -1,16 +1,50 @@
...
-    <Property name="root.log.level">INFO</Property>
+    <Property name="root.log.level">${env:LOG_LEVEL:-INFO}</Property>
+    <Property name="root.log.logger">${env:LOGGER:-Console}</Property>
+    <Property name="host">${env:HOST:-${docker:containerId:-localhost}}</Property>
+    <Property name="splunk.url">${env:SPLUNK_URL}</Property>
+    <Property name="splunk.token">${env:SPLUNK_TOKEN}</Property>
+    <Property name="splunk.index">${env:SPLUNK_INDEX}</Property>
+    <Property name="splunk.source">${env:SPLUNK_SOURCE:-besu}</Property>
+    <Property name="splunk.sourcetype">${env:SPLUNK_SOURCETYPE:-besu}</Property>
+    <Property name="splunk.batch_size_bytes">${env:SPLUNK_BATCH_SIZE_BYTES:-10000}</Property>
```
Comment (ziegfried) [inline besu/src/main/resources/log4j2.xml]:
```suggestion
    <Property name="splunk.batch_size_bytes">${env:SPLUNK_BATCH_SIZE_BYTES:-65536}</Property>
```

File: besu/src/main/resources/log4j2.xml [hunk 3]
```diff
@@ -1,16 +1,50 @@
...
+    <Property name="splunk.batch_size_count">${env:SPLUNK_BATCH_SIZE_COUNT:-100}</Property>
```
Comment (ziegfried) [inline besu/src/main/resources/log4j2.xml]:
```suggestion
    <Property name="splunk.batch_size_count">${env:SPLUNK_BATCH_SIZE_COUNT:-1000}</Property>
```

File: besu/src/main/resources/log4j2.xml [hunk 4]
```diff
@@ -1,16 +1,50 @@
...
+    <Property name="splunk.batch_interval">${env:SPLUNK_BATCH_INTERVAL:-5}</Property>
```
Comment (ziegfried) [inline besu/src/main/resources/log4j2.xml]:
```suggestion
    <Property name="splunk.batch_interval">${env:SPLUNK_BATCH_INTERVAL:-500}</Property>
```

File: besu/src/main/resources/log4j2.xml (hunk 1 above)
Comment (ziegfried) [reply besu/src/main/resources/log4j2.xml]: Added suggestion for new defaults.

**Pull request closed without merging**